shell.nix

```diff
@@ -3,8 +3,10 @@
 #pkgs.clangStdenv.mkDerivation {
 #  name = "libcxxStdenv";
 # clang_13
-
-pkgs.mkShell {
+let
+  inherit (pkgs.lib) strings;
+  jobs = if (strings.hasSuffix "linux" builtins.currentSystem) then "$(($(nproc)))" else "6";
```

Comment:
6 is just a magic value I used to leave a bit of room for other things on my 8-thread CPU while compiling Core. I don't have a strong preference to keep it - not sure what a sensible replacement would be

Reply:
Good to know. Yeah, I think one has to use something else than `nproc` on Mac for example. For Linux we could go with something like `nproc-2` combined with some `min()`-expression or whatever nix provides.

Reply:
For Mac we can copy this from @itornaza (thanks MIT license): https://github.com/itornaza/btc-dot/blob/master/bitcoin#L41

Reply:
Tentative solution: 0e8c532780376ebcb47f548889637e7094f75d76
User specifies `nix-shell --arg spareCores 2` in your case. No `max(a, b)` logic.

Reply:
Now covered in #7.